### DISCUSSION TITLE
Getting Started with Calliope & Energy Modelling

### DISCUSSION THREAD

**Discussion by ahkiat07** [calliope-project/calliope#661] [Q&A] [answered]
Hi Everyone,

I have successfully installed Calliope v0.7 as well as the CBC solver following the steps provided in the documentation. 
However, as I attempt to understand the first example/tutorial (national scale), I realise I have difficulty due to my limited knowledge on YAML syntax as well as energy modelling perhaps.

May I ask a few questions which hopefully could aid me in self-learning Calliope.
1. Is there a recommended YAML editor so that I could view/code with ease of identation & colors?
2. Is there any recommended self-guide and/or even online courses/workshops that will be useful in picking up all the relevant YAML basics & syntax? 
3. Is there any recommended self-help materials and/or online courses/workshops that will be useful in picking up energy modelling related knowledge?

Thanks everyone in advance!

**Comment by irm-codebase** [comment, marked as answer]
Hello @ahkiat07 !

Here are my recommendations for YAML:
- For all things YAML, read this: https://yaml.org/spec/1.2.2/
- I recommend using VSCode with the YAML extension: https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml

If you are using Github, you can use a pre-commit hook to check yaml syntax automatically:
- https://github.com/pre-commit/pre-commit-hooks (under `check-yaml`)

Best of luck!
